src/dbnode/encoding/proto/buffer.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// This file is mostly copy-pasta from: https://github.com/jhump/protoreflect/blob/master/dynamic/codec.go
+// since the jhump/protoreflect library does not expose the `codedBuffer` type.
```

Comment:
Hm, could we also translate the tests for this file?

Reply:
@robskillington The `protoreflect` library doesn't actually have tests for this file, I think in both cases it implicitly gets tested via other tests.

I could try and pull out the tests from the upstream protobuf library that the `protoreflect` ripped it out of, but I think we can delete this file altogether once this P.R lands: https://github.com/jhump/protoreflect/blob/master/dynamic/codec.go so maybe best to just wait until then (I need to upgrade the `protoreflect` library in a subsequent P.R anyways to pull in some perf improvements so I can wait until that P.R gets merged, update the dep, then delete this file)